### PR TITLE
Chapter 9

### PR DESCRIPTION
**Pull request by nguyenquoccuong1**
rubocop-chap9

### CODE REVIEW

**Comment by nguyenquoccuong1** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -13,7 +14,11 @@ def create
   end
 
   def destroy
-    log_out
+    log_out if logged_in?
     redirect_to root_url
   end
+
+  def check_remember user
+    params[:session][:remember_me] == "1" ? remember(user) : forget(user)
```
Comment (dothidiemthao) [inline app/controllers/sessions_controller.rb]: 1 settings nhé

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -3,15 +3,36 @@ def log_in user
...
   end
 
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
   def log_out
+    forget(current_user)
```
Comment (dothidiemthao) [inline app/helpers/sessions_helper.rb]: bỏ ( )

File: app/models/user.rb [hunk 3]
```diff
@@ -7,4 +8,34 @@ class User < ApplicationRecord
...
+
+  class << self
+    def digest string
+      cost = ActiveModel::SecurePassword.min_cost
+      if cost
+        BCrypt::Engine::MIN_COST
+      else
+        BCrypt::Engine.cost
+      end
+      BCrypt::Password.create(string, cost: cost)
```
Comment (dothidiemthao) [inline app/models/user.rb]: bỏ ( )

**Comment by dothidiemthao** [comment]
commented

**Comment by nguyenquoccuong1** [comment]
ready

**Review by dothidiemthao — APPROVED** [review]

**Pull request merged**